lib/ramble/ramble/definitions/families.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022-2025 The Ramble Authors
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+from typing import List
+
+
+class Families:
+    """Class representing object family membership"""
+
+    def __init__(
+        self,
+        origin_type: str,
+        families: List[str] = [],
+    ):
+        """Constructor for families object
+
+        Args:
+            origin_type: Base object type
+            families: List of families
+        """
+        self.family_type = f"{origin_type}_family"
+        self.families = families
+
```

Comment:
Maybe we could define `__iter__` here, and then the usage could be `for family in self.families` instead of `for family in self.families.families`.